widget/embedded/src/pages/AddCustomTokenPage.tsx

```diff
@@ -0,0 +1,192 @@
+      setSelectedBlockchainForCustomToken(null);
+      setAddress('');
+    }
+    setIsOpenErrorModal(false);
+  };
+
+  const handleErrorModalButtonClick = async () => {
+    if (error?.type === 'NETWORK_ERROR') {
+      await getCustomToken();
+    }
+    closeErrorModal();
+  };
+
+  useEffect(() => {
+    if (!!error) {
+      setIsOpenErrorModal(true);
+    }
+  }, [error]);
+  return (
+    <Layout
+      header={{
+        title: i18n.t('Add Custom Token'),
+      }}>
+      <PageContainer>
+        <Content>
+          <div>
+            <ItemPicker
+              onClick={() => navigate(navigationRoutes.blockchains)}
+              hasLogo={!!selectedBlockchainForCustomToken?.logo}
+              value={{
+                label: selectedBlockchainForCustomToken?.displayName,
+                logo: selectedBlockchainForCustomToken?.logo,
+              }}
+              title={i18n.t('Select chain')}
+              placeholder={i18n.t('Select chain')}
+            />
+            <Divider size={'24'} />
+            <Typography size="large" variant="label">
+              {i18n.t('Enter Address')}
+            </Typography>
+            <Divider size={10} />
+            <TextField
+              fullWidth
+              variant="contained"
+              placeholder={i18n.t('Enter token address')}
+              size="large"
+              value={address}
+              suffix={
+                !!address &&
+                isValidAddress && <DoneIcon color="success" size={12} />
+              }
+              onChange={(e) => setAddress(e.target.value)}
+            />
+            {!isValidAddress && !!address && (
+              <>
+                <Divider size={4} />
+                <Alert type="error" variant="alarm" title="Invalid Address" />
+              </>
+            )}
+          </div>
+
+          <Button
+            disabled={isImportDisabled}
+            type="primary"
+            variant="contained"
+            loading={loading}
+            size="large"
+            onClick={getCustomToken}>
+            {i18n.t('Import')}
+          </Button>
+        </Content>
+        <WatermarkedModal
+          open={isOpenErrorModal}
+          dismissible
+          onClose={closeErrorModal}
+          container={getContainer()}>
+          <MessageBox
+            title={error?.title || ''}
+            type="error"
+            description={error?.message}>
+            <Divider size={40} />
+            <Divider size={10} />
+
+            <Button
+              variant="contained"
+              size="large"
+              type="primary"
+              fullWidth
+              onClick={handleErrorModalButtonClick}>
+              {error?.type === 'NETWORK_ERROR'
+                ? i18n.t('Retry')
+                : i18n.t('Add another custom token')}
+            </Button>
+          </MessageBox>
+        </WatermarkedModal>
+        {!!selectedBlockchainForCustomToken && (
```

Comment:
I have two question here:
1. Why your modal is condional?
2. If it should be, why you didn't use `isOpenImportModa` instead of `selectedBlockchainForCustomToken` for your condition?

Reply:
There is no need to have a condition, I fixed it

Reply:
Since I wanted the `Blockchain` to be mandatory in the modal and the first `selectedBlockchainForCustomToken` is empty

Reply:
I deleted `isOpenImportModal`